firebase-firestore/src/main/java/com/google/firebase/firestore/FirestoreRegistrar.java

```diff
@@ -50,6 +51,7 @@ public List<Component<?>> getComponents() {
             .add(Dependency.optionalProvider(UserAgentPublisher.class))
             .add(Dependency.deferred(InternalAuthProvider.class))
             .add(Dependency.deferred(InternalAppCheckTokenProvider.class))
+            .add(Dependency.optionalProvider(MapEncoder.class))
```

Comment:
Using a provider like this is a potential future problem, I recommend using `Dependency.setOf()` instead as it will allow adding more MapEncoders in the future if needed. see [go/firebase-components-multibindings](http://go/firebase-components-multibindings) for details

Reply:
Thank you for your suggestion. I am using `Dependency.setof()` now. 
In the `DocumentSnapshot.set()` method, I am getting only the Kotlin `FirestoreMapEncoder` from the set of provided `MapEncoder` via a String `id` field.